components/merge-patient-lab-iips.js

```diff
@@ -14,12 +14,57 @@ var jswrapper = require('../src/javascript-wrapper');
  *
  * @param patient a JSON patient record to be processed
  * @param labwork a JSON lab record to be processed
- * @this component context
+ * @this vni context
  *
  * @return the combined patient/lab record
  */
-module.exports = jswrapper(function(patient, labwork) {
-    patient = _.isString(patient) ? JSON.parse(patient) : patient;
-    labwork = _.isString(labwork) ? JSON.parse(labwork) : labwork;
+module.exports = jswrapper( updater );
+
+/**
+ * Merges a patient and a labwork record into a single combined record.
+ * If no patient or labwork parameter is specified, the original parameter
+ * will be returned.
+ * 
+ * @this like all RDF pipeline updaters, this will usually be called on the 
+ *       vni context, however, this updater does not depend on that context. 
+ *
+ * @param patient a patient record in either json or string format
+ * @param labwork a lab record in either json or string format
+ * 
+ * @return the merged record
+ */
+function updater(patient, labwork) {
+
+    patient = jsObject( patient );
+    labwork = jsObject( labwork );
     return _.extend({}, patient, labwork);
-});
+}
+
+/**
+ * Converts whatever input parameter of any type (or even undefined) 
+ * into a javascript object 
+ * 
+ * @param parameter input parameter to be converted.  This could be
+ *        a string, already a javascript object, or even undefined.
+ *
+ * @return a javascript object 
+ */
+function jsObject( parameter ) { 
+
+    if ( _.isUndefined( parameter ) ) {  
+        return {}; 
+    } 
+
+    try { 
+        return  _.isString(parameter) ? JSON.parse(parameter) : parameter;
+    } catch ( e ) { 
+       var msg = "Unable to parse parameter "+parameter+": "+e;
+       console.error( msg );
+       throw Error( msg );
+    }
+} 
+
+// If running in test mode, mocha will execute internal unit testing
+if (process.env.NODE_ENV === 'test') {
+  module.exports._private = { updater: updater };
```

Comment:
In a previous meeting we discussed (and demonstrated) changing pipeline-component-factory.js to expose the updater function for testing, without a test flag.